File: app/src/main/java/com/king/mvvmframe/dao/AppDatabase.java
package com.king.mvvmframe.dao;


import com.king.mvvmframe.bean.SearchHistory;

import androidx.room.Database;
import androidx.room.RoomDatabase;

/**
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 */
@Database(entities = {SearchHistory.class}, version = 1,exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

//    public abstract SearchHistoryDao searchHistoryDao();
}

File: app/src/main/java/com/king/mvvmframe/data/AppDatabase.kt
package com.king.mvvmframe.data

import androidx.room.Database
import androidx.room.RoomDatabase
import com.king.mvvmframe.bean.SearchHistory
import com.king.mvvmframe.data.dao.SearchHistoryDao

/**
 * 数据库
 *
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 * <p>
 * <a href="https://github.com/jenly1314">Follow me</a>
 */
@Database(entities = [SearchHistory::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    /**
     * SearchHistoryDao
     */
    abstract val searchHistoryDao: SearchHistoryDao
}